pkg/user/signer.go

```diff
@@ -267,14 +267,30 @@ func (s *Signer) PubKey() cryptotypes.PubKey {
 	return s.pk
 }
 
+// Sequence returns the last signed sequence number of the signer
+func (s *Signer) Sequence() uint64 {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+	return s.lastSignedSequence
+}
+
 // GetSequence gets the latest signed sequence and increments the local sequence number
+// Deprecated: Use Sequence if you want to get the latest signed sequence number
 func (s *Signer) GetSequence() uint64 {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 	defer func() { s.lastSignedSequence++ }()
 	return s.lastSignedSequence
 }
 
+// incrementSequence gets the latest signed sequence and increments the local sequence number
+func (s *Signer) incrementSequence() uint64 {
```

Comment:
[Suggestion/Optional] I suggest renaming the function to `GetAndIncrementLastSequence`. This name provides a clearer understanding of the function's behavior.

Reply:
done although I kept the method private